Lib/fnmatch.py

```diff
@@ -77,24 +77,33 @@ def translate(pat):
     There is no way to quote meta-characters.
     """
 
-    STAR = object()
-    parts = _translate(pat, STAR, '.')
-    return _join_translated_parts(parts, STAR)
+    parts, indices = _translate(pat, '*', '.')
+    return _join_translated_parts(parts, indices)
 
+_re_setops_sub = re.compile(r'([&~|])').sub
+_re_escape = functools.lru_cache(maxsize=32768)(re.escape)
```

Comment:
Given this is only called to escape _single characters_, the maxsize seems OTT. I suggest something in the region of 256-1024.

Reply:
I think we should be fine with 512.